python/README.md

```diff
@@ -34,7 +34,41 @@ def do_something(params: str):
     ...
 ```
 
-At the moment, only functions that take 0 or 1 bytes arguments will work. These functions should also either return `None` or bytes.
+At the moment, only functions that take 0 or 1 `bytes` arguments will work. These functions should also either return `None` or `bytes`.
+
+#### Resource Requirements
+
+If your function has particular resource requirements, you should specify those during registration.
+
+For example, if your function needs 4 cores to run effectively, you can pass a configuration object.
+
+```python
+from preemo.worker import register
+
+@register(name="some_name", namespace="dev", cores=4)
+def do_something(params: bytes):
+    ...
+```
+
+You can specify the memory and/or storage requirments in either Mebibytes (`MiB`) or Gibibytes (`GiB`).
+
+```python
+from preemo.worker import register
+
+@register(name="some_name", namespace="dev", memory={ "MiB": 100 }, storage={ "GiB": 10 })
+def do_something(params: bytes):
+    ...
+```
+
+If you need GPUs to effectively run your function, you may pass the specific model to use.
+
+```python
+from preemo.worker import register
+
+@register(name="some_name", namespace="dev", gpu="nvidia-tesla-k80")
```

Comment:
Should we verify the passed GPU model? Does Google Cloud provide an instance with k80 GPU?

Reply:
The verification will all happen in the scheduler service

Reply:
How does the user know what kind of GPUs we support? Is there a way to let them know, without duplicating the code? For example, sent the enum values to the client and render it somewhere.

Reply:
I plan on making a python enum with all the options, for ease of use

Reply:
added a todo. Will take care of this in a subsequent PR